### PR TITLE
Make devices IDisposable

### PR DESCRIPTION
**Pull request by kayoub5**
Uses `IDisposable` to make tracking non-closed devices easier, in Code Analysis Tools
PR split into 3 commits for easier review.

### CODE REVIEW

File: SharpPcap/LibPcap/PcapDevice.cs [hunk 1]
```diff
@@ -225,6 +225,9 @@ public virtual PacketDotNet.LinkLayers LinkType
         /// </summary>
         public virtual void Close()
         {
+            //Remove event handlers
```
Comment (chmorgan) [inline SharpPcap/LibPcap/PcapDevice.cs:228]: why the move of this event handler being cleared out?

File: SharpPcap/LibPcap/PcapDevice.cs [hunk 2]
```diff
@@ -267,15 +258,8 @@ public virtual System.Net.NetworkInformation.PhysicalAddress MacAddress
         {
             get
             {
-                ThrowIfNotOpen("device not open");
```
Comment (chmorgan) [inline SharpPcap/LibPcap/PcapDevice.cs:270]: this check not necessary anymore?

**Review by chmorgan — CHANGES_REQUESTED** [review]
Possible to split up the Disposable change into smaller commits? Would make I easier from the git comments to see the why of some of the changes.

File: SharpPcap/LibPcap/PcapDevice.cs (hunk 1 above)
Comment (kayoub5) [reply SharpPcap/LibPcap/PcapDevice.cs:228]: Clearing of event handler won't take place if device has not been opened, or failed to open.

File: SharpPcap/LibPcap/PcapDevice.cs (hunk 2 above)
Comment (kayoub5) [reply SharpPcap/LibPcap/PcapDevice.cs:270]: Was not necessary to begin with, it's only using `PcapInterface.Addresses` and that does not require the device to be open.

**Comment by kayoub5** [comment]
> 
> 
> Possible to split up the Disposable change into smaller commits? Would make I easier from the git comments to see the why of some of the changes.

It's already 3 smallish commits:
* one for adding `IDisposable`
* one for changing target framework (so we could use [using statements](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement))
* one for replacing `Close` with `using device = ...`

**Review by chmorgan — APPROVED** [review]

**Pull request merged**